From 581bcb8c8802c2b763f5abc462a37e22e0195720 Mon Sep 17 00:00:00 2001
Message-Id: <581bcb8c8802c2b763f5abc462a37e22e0195720.1426116068.git.feitong.yi@intel.com>
From: Andrzej Datczuk <andrzej.datczuk@intel.com>
Date: Mon, 16 Feb 2015 14:01:15 +0100
Subject: [PATCH 1/5] SQUASHME! [VPG]: drm/i915: Perfmon disable RCS state
 check on load config

SQUASHME! - This patch should be squashed into the following existing
patch:

Author: Adam Rutkowski <adam.j.rutkowski@intel.com>
Date:   Wed Aug 27 13:52:22 2014 -0400
Change-Id: Ifc87f3542a8852943f13cf2bf724547df439b8d8
    REVERTME [VPG]: drm/i915: Perfmon OA/NOA configuration

Removed RCS state check for NULL during i915_perfmon_load_config
to prevent config load fails with deferred context initialization.
The mentioned state is not used in this and all called functions.

Issue: GMINL-3120
Change-Id: I6db75c33929484bac4162aeb8e969921d826f1c9
Signed-off-by: Andrzej Datczuk <andrzej.datczuk@intel.com>
---
 drivers/gpu/drm/i915/i915_perfmon.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_perfmon.c b/drivers/gpu/drm/i915/i915_perfmon.c
index 60b77b3..96d7c3f 100644
--- a/drivers/gpu/drm/i915/i915_perfmon.c
+++ b/drivers/gpu/drm/i915/i915_perfmon.c
@@ -511,7 +511,7 @@ static int i915_perfmon_load_config(
 				file_priv,
 				args->ctx_id);
 
-	if (IS_ERR_OR_NULL(ctx) || IS_ERR_OR_NULL(ctx->engine[RCS].state)) {
+	if (IS_ERR_OR_NULL(ctx)) {
 		DRM_DEBUG("invalid context\n");
 		ret = -EINVAL;
 		goto unlock_dev;
-- 
1.7.9.5

